FlowCrypt/Controllers/Attachment/AttachmentViewController.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  AttachmentViewController.swift
+//  FlowCrypt
+//
+//  Created by Evgenii Kyivskyi on 11/16/21
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+    
+import UIKit
+import WebKit
+import FlowCryptUI
+import Combine
+
+final class AttachmentViewController: UIViewController {
+
+    private lazy var webView = WKWebView()
+    private let file: FileType
+    private let shouldShowDownloadButton: Bool
+
+    private let filesManager: FilesManagerType
+    private lazy var attachmentManager = AttachmentManager(
+        controller: self,
+        filesManager: filesManager
+    )
+    
+    private var cancellable = Set<AnyCancellable>()
+
+    private let errorLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        label.text = "No preview available"
+        label.isHidden = true
+        label.textColor = .textColor
+        label.font = .systemFont(ofSize: 12)
+        return label
+    }()
+
+    init(
+        file: FileType,
+        shouldShowDownloadButton: Bool,
+        filesManager: FilesManagerType = FilesManager()
+    ) {
+        self.file = file
+        self.shouldShowDownloadButton = shouldShowDownloadButton
+        self.filesManager = filesManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationBar()
+        addWebView()
+
+        filesManager.save(file: file)
+            .sink(
+                receiveCompletion: {_ in},
+                receiveValue: { [weak self] url in
+                    self?.load(with: url)
+                }
+            )
+            .store(in: &cancellable)
```

Comment:
Here use a `Task` and call the async method with try/catch. On catch, handle the error.